qiskit/_result.py

```diff
@@ -85,20 +83,8 @@ def __iadd__(self, other):
         # TODO: reevaluate if moving equality to Backend themselves (part of
         # a bigger problem - backend instances will not persist between
         # sessions)
-        my_config = copy.deepcopy(self._qobj['config'])
-        other_config = copy.deepcopy(other._qobj['config'])
-        my_backend = my_config.pop('backend_name')
-        other_backend = other_config.pop('backend_name')
-
-        if my_config == other_config and my_backend == other_backend:
-            if isinstance(self._qobj['id'], str):
-                self._qobj['id'] = [self._qobj['id']]
-            self._qobj['id'].append(other._qobj['id'])
-            self._qobj['circuits'] += other._qobj['circuits']
-            self._result['result'] += other._result['result']
-            return self
-        else:
-            raise QISKitError('Result objects have different configs and cannot be combined.')
```

Comment:
Previously it checked that the 'config' section of the qobj matched, however, perhaps we can just match the backend_name since I don't think 'shots' or 'max_credits' mismatch should raise an error.